pkg/machinery/client/client.go

```diff
@@ -171,6 +173,14 @@ func (c *Client) GetConn(ctx context.Context, opts ...grpc.DialOption) (*grpc.Cl
 
 	var target string
 
+	for i, endpoint := range endpoints {
+		if net.AddressContainsPort(endpoint) {
+			endpoints[i] = net.FormatAddress(endpoint)
+		} else {
+			endpoints[i] = gonet.JoinHostPort(net.FormatAddress(endpoint), strconv.Itoa(constants.ApidPort))
```

Comment:
this is wrong for IPv6 - `net.FormatAddress` adds square brackets, and another square brackets in `JoinHostPort`, so just leave `net.JoinHostPort`